File: judge.cpp
#include <iostream>
#include <fstream>
#include <cstdio>
#include <cstdlib>
#include <cmath>
#include <cstring>
#include <string>
#include <algorithm>
#include <set>
#include <queue>
#include <vector>
#include <map>
#include <stack>
#include <time.h>

#include ".\user\example.cpp"

using namespace std;

int dataCount() {
	// Count the number of data
	fstream dataIn;
	dataIn.open("data");
	string dataLine;
	int dataCount = 0;
	while(getline(dataIn, dataLine))
		++dataCount;
	dataIn.close();
	return dataCount;
}

void userRun(int dataCount) {
	// Redirect cin and cout
	freopen("data","r",stdin);
	freopen("out","w",stdout);

	// Run the user program
	Solution sol;
	for(int i = 0; i < dataCount; ++i)
		sol.solve();

	// Redirect cout back to the console
	freopen("CON", "w", stdout);
}

int judge() {
	// Open ans and out files
	fstream stdAns;
	stdAns.open("answer", ios::in);
	fstream usrAns;
	usrAns.open("out", ios::in);

	// Count the number of standard answer and user answer
	string stdAnsLine, usrAnsLine;
	int stdCount = 0, usrCount = 0;
	while(getline(stdAns, stdAnsLine)) {
		++stdCount;
	}
	while(getline(usrAns, usrAnsLine)) {
		++usrCount;
	}

	// Reposition the file pointer
	stdAns.clear();
	stdAns.seekg(0, ios::beg);
	usrAns.clear();
	usrAns.seekg(0, ios::beg);

	// Compare the answers
	int corCount = 0;
	for(int i = 0; i < min(stdCount, usrCount); ++i) {
		getline(stdAns, stdAnsLine);
		getline(usrAns, usrAnsLine);
		// cout << stdAnsLine << "   " << usrAnsLine << endl;
		if(stdAnsLine == usrAnsLine)
			++corCount;
		else
			break;
	}

	// Close all the files
	stdAns.close();
	usrAns.close();

	// Return the result
	return int(100 * corCount / (double)stdCount);
}

int main() {
	userRun(dataCount());
	cout << "Pass Rate : " << judge() << " % " << endl;
	return 0;
}

File: judge.py
import math
from math import *
from user.example import *


# Count how many lines in data
def dataCount():
    dataIn = open("data", "r")
    dataCount = len(dataIn.readlines())
    dataIn.close()
    return dataCount


def userRun(dataCount):
    # Redirect data input and user output
    import sys
    dataIn = open("data", "r")
    usrOut = open("out", "w")
    stdInBackup = sys.stdin
    stdOutBackup = sys.stdout
    sys.stdin = dataIn
    sys.stdout = usrOut

    # Run user program
    sol = Solution()
    for i in range(dataCount):
        sol.solve()

    # Restore stdin and stdout
    sys.stdin = stdInBackup
    sys.stdout = stdOutBackup

    # Close the files
    dataIn.close()
    usrOut.close()


def judge():
    # Open user output and answer
    stdAns = open("answer", "r")
    usrAns = open("out", "r")

    # Read in the answer and user output
    stdAnsList = stdAns.readlines()
    usrAnsList = usrAns.readlines()

    # Close the files
    stdAns.close()
    usrAns.close()

    # Compare the files
    corCount = 0
    for i in range(min(len(stdAnsList), len(usrAnsList))):
        if stdAnsList[i] == usrAnsList[i]:
            corCount += 1
        else:
            break

    return math.floor(100 * corCount / len(stdAnsList))


def main():
    userRun(dataCount())
    print("Pass Rate : " + str(judge()) + " % ")


if __name__ == "__main__":
    main()
